### ISSUE TITLE
bthread::TaskGroup::sched_to() function crash

### ISSUE THREAD

**Issue by zorrofox** [StarRocks/starrocks#20677]
### Steps to reproduce the behavior (Required)
1. CREATE TABLE sbtest1
2. CREATE ROUTINE LOAD kafka_sbtest.kafka_sbtest1 ON sbtest1 from a Kafka topic
And the BE will crash to restart

### Expected behavior (Required)
No crash and the ROUTINE LOAD successfully
### Real behavior (Required)
The BE crash and Dump core bellow:
```
start time: Thu Mar 30 11:54:06 UTC 2023
*** Check failure stack trace: ***
query_id:00000000-0000-0000-0000-000000000000, fragment_instance:00000000-0000-0000-0000-000000000000
tracker:process consumption: 115552776
tracker:query_pool consumption: 0
tracker:load consumption: 0
tracker:metadata consumption: 453674
tracker:tablet_metadata consumption: 303786
tracker:rowset_metadata consumption: 149888
tracker:segment_metadata consumption: 0
tracker:column_metadata consumption: 0
tracker:tablet_schema consumption: 17786
tracker:segment_zonemap consumption: 0
tracker:short_key_index consumption: 0
tracker:column_zonemap_index consumption: 0
tracker:ordinal_index consumption: 0
tracker:bitmap_index consumption: 0
tracker:bloom_filter_index consumption: 0
tracker:compaction consumption: 0
tracker:schema_change consumption: 0
tracker:column_pool consumption: 0
tracker:page_cache consumption: 0
tracker:update consumption: 0
tracker:chunk_allocator consumption: 0
tracker:clone consumption: 0
tracker:consistency consumption: 0
*** Aborted at 1680177402 (unix time) try "date -d @1680177402" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x9d) received by PID 157 (TID 0xfffe9f8c7c80) from PID 157; stack trace: ***
    @          0x4829130 google::(anonymous namespace)::FailureSignalHandler()
    @     0xffffb83aa860 ([vdso]+0x85f)
    @     0xffffb6f6f200 (unknown)
    @     0xffffb6f2a67c raise
    @     0xffffb6f17130 abort
    @          0x1eb3dd0 starrocks::failure_function()
    @          0x481be88 google::LogMessage::Fail()
    @          0x481e18c google::LogMessage::SendToLog()
    @          0x481b9f4 google::LogMessage::Flush()
    @          0x481e680 google::LogMessageFatal::~LogMessageFatal()
    @          0x495af94 bthread::TaskGroup::sched_to()
    @          0x495c250 bthread::TaskGroup::ending_sched()
    @          0x495ce88 bthread::TaskGroup::task_runner()
    @          0x4aad9cc bthread_make_fcontext
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN 46bf084a7 |
+-------------------+

**This SR is compiled in an ARM64 docker by myself used v2.5.3 and running on a Kubernetes cluster by the Operator.**

**Comment by jaogoy** [comment]
@rickif have a look please.

**Comment by rickif** [comment]
@zorrofox Could you please provide the last several log messages in be.INFO?

**Comment by stdpain** [comment]
fixed by #16418
